packages/terra-date-time-picker/src/DateTimeUtils.js

```diff
@@ -62,10 +62,15 @@ class DateTimeUtils {
    */
   static syncDateTime(momentDate, iOSdate, time) {
     const date = moment(iOSdate);
-    let newDate = momentDate ? momentDate.clone() : date;
 
-    // If momentDate was null, a new moment date needs to be created and sync'd with the entered time.
-    if (momentDate === null && time && time.length === 5) {
+    if (!date.isValid()) {
+      return momentDate;
```

Comment:
Should we be checking that the momentDate is valid here like we are below? `momentDate.isValid()`

Reply:
This is to return the passed in moment object in the event that the iSOdate is invalid. I have made some updates to this function. https://github.com/cerner/terra-framework/pull/619/commits/784e52a725189be07a1cc7da27ca567b630385ed. See above comment for details.